matplotlib2tikz/path.py

```diff
@@ -129,30 +133,36 @@ def draw_pathcollection(data, obj):
         else:
             draw_options.append('colormap/' + mycolormap)
 
-    if len(obj.get_sizes()) == len(dd):
-        # See Pgfplots manual, chapter 4.25.
-        # In Pgfplots, \mark size specifies raddi, in matplotlib circle areas.
-        radii = numpy.sqrt(obj.get_sizes() / numpy.pi)
-        dd = numpy.column_stack([dd, radii])
-        labels.append('sizedata' + 14*' ')
-        draw_options.extend([
-            'visualization depends on='
-            '{\\thisrow{sizedata} \\as\\perpointmarksize}',
-            'scatter/@pre marker code/.append style='
-            '{/tikz/mark size=\\perpointmarksize}',
-            ])
-
-    do = ' [{}]'.format(', '.join(draw_options)) if draw_options else ''
-    content.append('\\addplot{}\n'.format(do))
-
-    to = ' [{}]'.format(', '.join(table_options)) if table_options else ''
-    content.append('table{}{{%\n'.format(to))
-
-    content.append((' '.join(labels)).strip() + '\n')
-    fmt = (' '.join(dd.shape[1] * ['%+.15e'])) + '\n'
-    for d in dd:
-        content.append(fmt % tuple(d))
-    content.append('};\n')
+    num_paths = len(obj.get_paths()) if is_contour else 1
+
+    for i in range(num_paths):
+        if is_contour:
+            dd = obj.get_paths()[i].vertices
```

Comment:
The `is_contour` logic is quite weird in the method; this needs cleaning up. For example: Why does `dd` need to get overridden here?

Reply:
I couldn't find another way. If there are more than one contour bands with same value, they are collected in more than one path in an obj. The solution I use it to output `\addplot` for each of the paths, so we need to iterate through paths. For each iteration a new set of points for `table` is needed, therefore `dd` is redefined for each path. Conditional is needed to keep original behavior for other types of plot (let's say scatter plot). If conditional is removed test_scatter will definitely fail.